LibraBFT/Impl/Consensus/SafetyRules/SafetyRules.agda

```diff
@@ -47,6 +44,23 @@ extensionCheckM voteProposal = do
 
 ------------------------------------------------------------------------------
 
+constructLedgerInfoM : Block → HashValue → LBFT (Either ErrLog LedgerInfo)
+constructLedgerInfoM proposedBlock consensusDataHash = do
+  let block2 = proposedBlock ^∙ bRound
+      block1 = proposedBlock ^∙ bQuorumCert ∙ qcCertifiedBlock ∙ biRound
+      block0 = proposedBlock ^∙ bQuorumCert ∙ qcParentBlock ∙ biRound
+      commit = (block0 + 1 == block1) ∧ (block1 + 1 == block2)
+  commitInfo ←
+    if commit
+    then (do
+      let c = proposedBlock ^∙ bQuorumCert ∙ qcParentBlock
+      pure c)
```

Comment:
Similar comment as above about logging.

Reply:
BTW, I note that we're not modeling our enter/exit logging used for debugging, and I am fine with that, including long term.

Reply:
DONE: `logInfo` added